File: SelectionSort.java
public class SelectionSort
{
    private int[] array;
    public SelectionSort(int[] array)
    {
        this.array = array;
    }
    public void sort()
    {
        System.out.println("Sorting");
        //Iterates through the array
        for(int i = 0; i < size; i++)
        {
            int minIndex = i;
            //Find the smallest element
            for(int j = i+1; j < size; j++)
            {
                if(array[minIndex] > array[j])
                {
                    minIndex = j;
                }
            }
            //Swap the smallest element and the current elemtn
            int temp = array[minIndex];
            array[minIndex] = array[i];
            array[i] = temp;
        }
    }
}

File: selectionsort.js
function SelectionSort(input) {  
    for (var i = 0; i < input.length; i++) {  
        var temp = input[i];  
        for (var j = i + 1; j < input.length; j++) {  
            if (temp > input[j]) {  
                temp = input[j];  
            }  
        }  
        var index = input.indexOf(temp);  
        var tempVal = input[i];  
        input[i] = temp;  
        input[index] = tempVal;  
    }  
}  
